spec/filters/json_spec.rb

```diff
@@ -224,6 +226,41 @@
     end
   end
 
+  describe "target" do
+
+    let(:logger) { filter.logger }
+
+    before { allow( logger ).to receive(:info) }
+
+    context 'not set in ECS mode' do
+      let(:config) { { "source" => "message", 'ecs_compatibility' => 'v1' } }
+      let(:message) { ' { "foo": "bar" } ' }
+
+      it "works but logs a warning" do
+        filter.register
+        expect( logger ).to have_received(:info).with(/ECS compatibility is enabled but `target` option was not specified/i)
+
+        filter.filter(event)
+        expect( event.get('foo') ).to eql 'bar'
+      end
+    end
+
+    context 'set in ECS mode' do
+      let(:config) { { "source" => "message", 'ecs_compatibility' => 'v1', 'target' => '[baz]' } }
+      let(:message) { ' { "foo": "bar" } ' }
+
+      it "works (no warning logged)" do
+        filter.register
+        expect( logger ).to_not have_received(:info)
```

Comment:
If we _do_ validate this behaviour (as opposed to relying on the upstream validation as suggested in another comment), the use of `Logger#info` alone shouldn't be enough to make this spec fail. We should refine our assertion to capture only infos that are directly related to what we are validating.